sdk/storage/sample/datalake_getting_started.cpp

```diff
@@ -3,11 +3,80 @@
 
 #include "datalake/datalake.hpp"
 #include "samples_common.hpp"
+#include <iostream>
 
 SAMPLE(DataLakeGettingStarted, DataLakeGettingStarted)
 void DataLakeGettingStarted()
 {
+  // Please note that you can always reference test cases for advanced usages.
+
   using namespace Azure::Storage::Files::DataLake;
-  auto client = ServiceClient::CreateFromConnectionString(GetConnectionString());
-  auto response = client.ListFileSystems();
+
+  std::string fileSystemName = "sample-file-system";
+  std::string directoryName = "sample-directory";
+  std::string fileName = "sample-file";
+
+  // Initializing a ServiceClient that can then initialize the FileSystemClient or list file
+  // systems.
+  auto serviceClient = ServiceClient::CreateFromConnectionString(GetConnectionString());
+
+  // List all file systems.
+  std::string continuation;
+  std::vector<FileSystem> fileSystems;
+  do
+  {
+    auto response = serviceClient.ListFileSystems();
+    if (response->Continuation.HasValue())
+    {
+      continuation = response->Continuation.GetValue();
+    }
+    fileSystems.insert(
+        fileSystems.end(), response->Filesystems.begin(), response->Filesystems.end());
+  } while (!continuation.empty());
```

Comment:
This should go after create filesystem

Reply:
I don't think so, the sequence is sample for ServiceClient->FileSystemClient->Path/Directory/FileClient. I think it is better.

Reply:
CreateFilesystem is very basic, ListFilesystem is more advanced. Plus the list result is very likely to be empty.

Reply:
Yes, but among all the other things in this sample, this is the one that gets asked most in Emails/Github issues from my feeling.

Reply:
I cannot remember, can you share some of those Emails/GitHub issues?

Even if it was true, it still couldn't justify that ListFileSystem is a more important API than CreateFileSystem and it should come first.

Reply:
I believe you just handled a case from Nara V and we have databox reporting similar issues of listing returning empty value.

Reply:
So some users have some questions about a specific API, so what? We should put every API that some customer ever asked before at the very beginning of this sample?

Reply:
First, it is already not at the beginning of the latest commit, since I was out-voted for what is the most important in the sample. Second, I want to avoid answering the questions we keep getting asked for track 1 SDK by putting it into the most obvious places in the sample, I don't feel wrong for it.
Also, I never encountered even one occasion where a customer complains about the sample containing too much information, it's usually the other way around where customers complain about the sample not containing enough information.